tardis/energy_input/samplers.py

```diff
@@ -138,3 +140,42 @@ def sample_decay_time(
             np.random.random()
         )
     return decay_time
+
+
+class PositroniumSampler:
+    def __init__(self, n_grid=1000):
+        """
+        Parameters
+        ----------
+        n_grid : int, optional
+            Number of grid points for the CDF, by default 1000
+        """
+        self.x_grid = np.linspace(0.01, 0.99, n_grid)
+        self.cdf_grid = np.array(
+            [
+                np.trapz(self.pdf(self.x_grid[:i]), self.x_grid[:i])
+                for i in range(len(self.x_grid))
+            ]
+        )
+        self.cdf_grid /= self.cdf_grid[-1]
+
+    @staticmethod
+    def pdf(x):
```

Comment:
Could this function get a source added? As is I'm not sure where this PDF comes from and why it looks like this.